a10_octavia/controller/worker/tasks/cert_tasks.py

```diff
@@ -24,182 +24,200 @@
 LOG = logging.getLogger(__name__)
 
 
+class GetSSLCertData(task.Task):
+    """Task to get SSL certificate data from Barbican"""
+
+    def execute(self, loadbalancer, listener):
+        cert_data = None
+        if listener.protocol == 'TERMINATED_HTTPS' and listener.tls_certificate_id:
+            try:
+                barbican_client = BarbicanACLAuth().get_barbican_client(loadbalancer.project_id)
+                cert_data = utils.get_cert_data(barbican_client, listener)
+                # Monkey patching template name as listener ID
```

Comment:
This isn't really monkey patching. You aren't replacing/adding a class function or anything.